src/SignatureSectionProcessor.h

```diff
@@ -371,13 +371,15 @@ namespace scpl {
          * \param begin Character index representing the beginning of the bracket that needs to be matched
          * \param endBracket The type of bracket that needs to be matched
          * \param splitByAttribute If this is true, we need to return when we find a top-level attribute delimiter
+         * \param clearAtEnd If this is true, the string will be cleared at the end of parsing
```

Comment:
This function feels overloaded with features, in a future it may be a good idea to break it into two parts.